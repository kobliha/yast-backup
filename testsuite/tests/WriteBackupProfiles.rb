# encoding: utf-8

# Backup module - testsuite
#
# WriteBackupProfiles function test
#
# testedfiles: Backup.ycp
#
# $Id$
#
module Yast
  class WriteBackupProfilesClient < Client
    def main
      Yast.import "Testsuite"

      @readmap = {
        "target" => { "string" => "", "size" => -1, "tmpdir" => "/tmp" },
        "proc"   => { "cpuinfo" => { "value" => { "0" => { "flags" => "" } } } },
        "probe"  => {
          "has_smp" => false,
          "is_uml"  => false,
          "cdrom"   => { "manual" => [] }
        }
      }

      @execmap = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stderr" => "",
            "stdout" => "charmap=\"ISO-8859-1\"\n"
          }
        }
      }

      Testsuite.Init([@readmap, {}, @execmap], nil)

      Yast.import "Pkg" # override packamanager

      Yast.import "Backup"

      Testsuite.Dump(Backup.backup_profiles)

      Backup.StoreSettingsToBackupProfile("default profile")

      # setup global settings
      Backup.archive_name = "Does not work"
      Backup.description = "Does not work - profile"
      Backup.archive_type = nil
      Backup.multi_volume = nil
      Backup.volume_size = nil
      Backup.user_volume_size = nil
      Backup.user_volume_unit = nil
      Backup.do_search = nil
      Backup.system = nil
      Backup.display = nil
      Backup.do_md5_test = nil
      Backup.default_dir = nil
      Backup.dir_list = nil
      Backup.fs_exclude = nil
      Backup.detected_fs = nil
      Backup.detected_ext2 = nil
      Backup.ext2_backup = nil
      Backup.backup_pt = nil
      Backup.backup_all_ext2 = nil
      Backup.backup_none_ext2 = nil
      Backup.backup_selected_ext2 = nil
      Backup.unselected_files = nil

      Backup.selected_files = nil
      Backup.backup_files = {}

      Backup.StoreSettingsToBackupProfile("empty profile")

      Testsuite.Dump(Backup.backup_profiles)

      nil
    end
  end
end

Yast::WriteBackupProfilesClient.new.main
