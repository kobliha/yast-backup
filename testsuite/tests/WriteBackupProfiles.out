Dump	$[]
Dump	$["default profile":$[`archive_name:"", `archive_type:`tgz, `autoprofile:true, `backup_all_ext2:false, `backup_none_ext2:true, `backup_pt:true, `backup_selected_ext2:false, `default_dir:["/media", "/tmp", "/var/lock", "/var/run", "/var/tmp", "/var/cache", "/sys", "/windows"], `description:"", `detected_ext2:nil, `detected_fs:nil, `dir_list:["/media", "/tmp", "/var/lock", "/var/run", "/var/tmp", "/var/cache", "/sys", "/windows"], `display:false, `do_md5_test:true, `ext2_backup:[], `fs_exclude:["iso9660", "ntfs"], `include_dirs:["/"], `mail_summary:true, `multi_volume:false, `nfsexport:"", `nfsserver:"", `perms:true, `regexp_list:[], `search:true, `system:false, `target_type:`file, `tmp_dir:"/tmp", `unselected_files:[], `user_volume_size:"", `user_volume_unit:nil, `volume_size:`fd144], "empty profile":$[`archive_name:"Does not work", `archive_type:nil, `autoprofile:true, `backup_all_ext2:nil, `backup_none_ext2:nil, `backup_pt:nil, `backup_selected_ext2:nil, `default_dir:nil, `description:"Does not work - profile", `detected_ext2:nil, `detected_fs:nil, `dir_list:nil, `display:nil, `do_md5_test:nil, `ext2_backup:nil, `fs_exclude:nil, `include_dirs:["/"], `mail_summary:true, `multi_volume:nil, `nfsexport:"", `nfsserver:"", `perms:true, `regexp_list:[], `search:nil, `system:nil, `target_type:`file, `tmp_dir:"/tmp", `unselected_files:nil, `user_volume_size:nil, `user_volume_unit:nil, `volume_size:nil]]
